packages/cardano-services-client/src/ChainHistoryProvider/BlockfrostChainHistoryProvider.ts

```diff
@@ -34,6 +34,7 @@ type BlockfrostTx = Pick<Responses['address_transactions_content'][0], 'block_he
 const compareTx = (a: BlockfrostTx, b: BlockfrostTx) => a.block_height - b.block_height || a.tx_index - b.tx_index;
 
 export class BlockfrostChainHistoryProvider extends BlockfrostProvider implements ChainHistoryProvider {
+  private readonly cache: Map<string, Cardano.HydratedTx> = new Map();
```

Comment:
It would be better if cache objects were injected (e.g. `type Cache<T> = { get: (key) => Promise<T>, set: (key: string, value: T) => Promise<void> }`), then it would be easier to update/swap it with a more sophisticated cache (lru, persistent etc.) and control it with FFs. Anyway happy to merge as-is to mitigate the current issue quickly.